veps/vep-1.md

```diff
@@ -0,0 +1,108 @@
+SELECT 1;
+```
+
+Vitess `4.xx` **MUST** still support this query, because blocking it would be backwards incompatible. However, Vitess `4.xx` **MAY** add an option to _prevent unsafe queries_. It is expected that the default behavior of _prevent unsafe queries_ would be `FALSE`. Executing the query **SHOULD** result in a deprecation warning.
+
+When Vitess `5.xx` is released, the default for _prevent unsafe_queries_ **MAY** be changed to `TRUE`. Waiting until Vitess `5.xx` to change the default makes the behavior _forwards compatible_ because users can test out the change prior to upgrading. Attempting to execute the query **SHOULD** produce an error stating that the `4.xx` behavior can be restored by changing a setting, but the setting is deprecated and will be removed soon.
+
+When Vitess `6.xx` is released, the option to _prevent unsafe queries_ **MAY** be removed with the implied behavior that the option is permanently `TRUE`.
+
+_Based on the impact of functionality being removed, the Vitess team **MAY** at their discretion extend the deprecation cycle. This example serves to demonstrate the minimum time required._
+
+### Example 2: Deprecating and removing an obsolete utility
+
+Version `3.xx` of Vitess supports a utility that has been made obsolete by recently added functionality:
+
+```
+vtdescribe: an obsolete utility
+```
+
+Vitess `4.xx` **SHOULD** invoke a warning when attempting to use this utility, with the utility completely removed in Vitess `5.xx`.
+
+_This change is backward compatible in Vitess `4.xx` because the utility can still be used. It is forwards compatible because attempting to use it will suggest the alternative functionality which is available._
+
+## Exclusions
+
+The following is not subject to the Backwards and Forwards Compatibility Promise:
+
+* Functionality which has been marked as experimental. As new features are developed, the Vitess authors may release it in a stable release but as experimental. As feedback is received, changes may be made to the feature without respecting the regular deprecation cycle associated with backwards and forwards compatibility.
+
+* Functionality limited to the source code and/or build environment. If the Vitess authors desired to make a change, such as upgrading the required Go version, or switching from `govendor` to Go modules, there is no requirement to ensure compatibility with older versions of Go. The promise is intended for user facing functionality.
+
+## Support Lifecycle
+
+Because the release cycle includes a new release every 12 weeks, most fixes will only be pushed to the master branch. An exception to this process will be made for high severity bugs, in which case a fix will be available in the weekly release for all currently supported versions.
+
+High severity bugs in this context include CVEs, data corruption, wrong results or outage inducing issues. The Vitess team **MAY** decide to only backport bugs with higher user impact.
```

Comment:
I would add a link to [CVEs](https://cve.mitre.org/) - to be completely explicit.